### ISSUE TITLE
in future add integration with Towny

### ISSUE THREAD

**Issue by ColdTearsYY** [ItzAmirreza/EzChestShop#50]
Plugin: Towny [https://github.com/TownyAdvanced/Towny](url)
Everyone can create a chestshop in other's town plot through an existing chest here. Please check the permission before people could create a chestshop.

**Comment by ElitoGame** [comment]
Already on the TODO list ^^
Though I am still curious why it doesn't work naturally, but I also have never used Towny before 🤔

**Comment by ColdTearsYY** [comment]
Is it has something to do with the loading sequence? (Guessing...)

**Comment by ElitoGame** [comment]
No, the question is: can people build in other towns? We are checking if players can break blocks in an area in order to determine if they can create a shop there. So why does this check return true there?

**Comment by vincenzodago** [comment]
Hi.
If the mayor of the city sets town/plot perms in a certain way then even players not belonging to his city can also build.
Anyway, I think this method is right for you :
https://github.com/TownyAdvanced/Towny/blob/22251f29ed86d3a1e25db1e9cf9cb56e66e98a61/Towny/src/main/java/com/palmergames/bukkit/towny/utils/ShopPlotUtil.java#L57
ECS checks if it is a plot shop and if the player has permission to build there.

**Comment by ElitoGame** [comment]
I did write the code, just need to set Towny up and test it. Join our discord and I'll provide a beta build there, as soon as I am kinda sure it works.

**Comment by ElitoGame** [comment]
I provided a beta build on discord - check the suggestions forum > Towny plugin support. Please test it in a dev environment, I am very unexperienced with Towny and I only had 1 account available for testing. Let me know how it goes on Discord ^^
